learningobjects/api.py

```diff
@@ -0,0 +1,39 @@
+"""
+Helper functions for using the models, so external
+apps don't tie functionality to internal implementation.
+"""
+
+from datetime import datetime
+
+from learningobjects.models import Course, Repository
+
+# pylint: disable=no-member
+
+def create_course(org, course_number, semester, user_id):
+    """
+    Add a course to the database.
+    """
+    # TODO: Determine rules for accepting/requiring a repository.
+    kwargs = {
+        "org": org, "course_number": course_number, "semester": semester,
+        "import_date": datetime.now(), 'imported_by_id': user_id,
+        "repository_id": get_temp_repository(user_id).id
+    }
+    courses = Course.objects.filter(**kwargs)
```

Comment:
But you are doing the get with your filter here, you can always ignore the object created if created is false, i.e.:
`course, created = ...get_or_create(...)`
`if created: return object; else raise Course.ValidationError`
I also think you should raise instead of tuple as was discussed somewhat in chat